### ISSUE TITLE
optuna.integration.lightgbm add learning_rate and n_estimators

### ISSUE THREAD

**Issue by mirix** [optuna/optuna#4618]
### Motivation

optuna.integration.lightgbm provides a very convenient approach for hyperparameter optimisation

However, it does not not seem to optimise some of the most determinant hyperparameters such as learning_rate and n_estimators.

The module would become more useful it it would enable one to optimise the most relevant LightGBM hyperparameters. 

### Description

optuna.integration.lightgbm add learning_rate and n_estimators

Or is there any technical reason why these two crucial hyperparameters are not being currently optimised?

**Comment by contramundum53** [comment]
Thanks for creating an issue! (and sorry for replying late 🙇‍♂️ )
I think this can be fixed relatively easily by modifying https://github.com/optuna/optuna/blob/master/optuna/integration/_lightgbm_tuner/optimize.py .
I'm not currently having enough time, but we appreciate your contributions!

**Comment by nabenabe0928** [comment]
Hi @mirix, thank you for raising this point:)

In a nutshell, `learning_rate` and `n_estimators` (`num_boost_round`) are not tuned on purpose.
Talking about the background story, our `LightGBMTuner` is based on [the tutorial by Kaggle Grandmaster](https://www.youtube.com/watch?v=VC8Jc9_lNoY&t=1018s). 

In essence, it is empirically known among Kagglers that the performance of LightGBM improves when we train it for a longer time, i.e. to add more estimators and to take a large `num_boost_round`, with a small `learning_rate`.
In other words, `n_estimators` or `num_boost_round` corresponds to the training epoch size in deep learning, which is a budget, but not a hyperparameter.

Then the obvious question is how we should choose `learning_rate`.
In this regard, competitive Kagglers fix this parameter to a well-used value rather than tuning to avoid overfitting the validation dataset.
Technically speaking, we can definitely tune `learning_rate`, but we did not dare to do it to avoid overfitting.
